controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -0,0 +1,213 @@
+)
+
+type Query struct {
+	Title string `json:"title"`
+	Query string `json:"query"`
+}
+
+type Config struct {
+	Title   string  `json:"title"`
+	Type    string  `json:"type"`
+	Queries []Query `json:"queries"`
+}
+
+type MetricsDashboardConfigMapData struct {
+	Data []Config `json:"data"`
+}
+
+var _ SubResourceReconciler = (*KserveMetricsDashboardReconciler)(nil)
+var ovmsData []byte
+var tgisData []byte
+var vllmData []byte
+
+type KserveMetricsDashboardReconciler struct {
+	NoResourceRemoval
+	client           client.Client
+	telemetryHandler resources.ConfigMapHandler
+	deltaProcessor   processors.DeltaProcessor
+}
+
+func NewKserveMetricsDashboardReconciler(client client.Client) *KserveMetricsDashboardReconciler {
+	return &KserveMetricsDashboardReconciler{
+		client:           client,
+		telemetryHandler: resources.NewConfigMapHandler(client),
+		deltaProcessor:   processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveMetricsDashboardReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+
+	// resolve SR
+	isvcRuntime := isvc.Spec.Predictor.Model.Runtime
+	runtime := &kservev1alpha1.ServingRuntime{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: *isvcRuntime, Namespace: isvc.Namespace}, runtime); err != nil {
+		log.Error(err, "Could not determine servingruntime for isvc")
+	}
+
+	servingRuntimeImage := runtime.Spec.Containers[0].Image
+	re := regexp.MustCompile(`/([^/@]+)[@:]`)
+	findImageName := re.FindStringSubmatch(servingRuntimeImage)
+	servingRuntime := findImageName[1]
+
+	var data []byte
+	switch servingRuntime {
+	case "openvino_model_server":
+		if ovmsData == nil {
+			ovmsData, err := os.ReadFile("ovms-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+				return nil, err
+			}
+			data = ovmsData
+		}
+	case "text-generation-inference":
+		if tgisData == nil {
+			tgisData, err := os.ReadFile("tgis-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+				return nil, err
+			}
+			data = tgisData
+		}
+
+	case "vllm":
+		if vllmData == nil {
+			vllmData, err := os.ReadFile("vllm-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+				return nil, err
+			}
+			data = vllmData
+		}
```

Comment:
we need a default case here for unsupported runtimes
```suggestion
		}
	default:
		log.V(1).Info("Metrics for runtime not supported.", "Runtime:", runtimeImageName)
		configMap := &corev1.ConfigMap{
			ObjectMeta: metav1.ObjectMeta{
				Name:      isvc.Name + ConfigMapSuffix,
				Namespace: isvc.Namespace,
			},
			Data: map[string]string{
				"supported": "false",
			},
		}
		if err := ctrl.SetControllerReference(isvc, configMap, r.client.Scheme()); err != nil {
			log.Error(err, "Unable to add OwnerReference to the Metrics Dashboard Configmap")
			return nil, err
		}
		return configMap, nil
```